openquake/calculators/extract.py

```diff
@@ -952,7 +952,12 @@ def extract_gmf_npz(dstore, what):
         # zero GMF
         yield 'rlz-%03d' % rlzi, []
     else:
-        gmfa = _gmf(df, n, oq.imtls)
+        imts = [imt for imt in oq.imtls]
+        try:
+            sec_imts = oq.sec_imts
+        except AttributeError:
+            sec_imts = []
```

Comment:
I believe it is impossible to get an AttributeError here. How did you get it?

Reply:
That error never occurred to me, but I was cautiously assuming it might be possible that oq.sec_imts is not defined (for instance for old calculations). If we are sure that oq.sec_imts is always available, I can avoid that try-except.

Reply:
You should have grepped for `sec_imts` and you would have discovered that it is always defined.